src/chromeasync.js

```diff
@@ -0,0 +1,59 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Wraps a chrome API, which takes the callback with an argument at the last
+ * argument, * and returns a function returning the Promise instance.
+ * Returned promise will be rejected if wrapped API succeeds (i.e.
```

Comment:
Hmm, the explanation here sounds not true? I guess:

A returned promise will be resolved with the return value if the API call succeeds. Otherwise, it is rejected with chrome.runtime.lastError value.